### PR TITLE
upgrade kafka-connect to 2.4.1 in compile-time dependency while ensur…

### PR DESCRIPTION
**Pull request by elakito**
…ing runtime compatibility with kafka-connect 2.2.x, 2.3.x

Resulted from this question.
 https://github.com/Apicurio/apicurio-registry/issues/557

### CODE REVIEW

File: search/connector/src/main/java/io/apicurio/registry/connector/Compatibility.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 Red Hat
```
Comment (alesj) [inline search/connector/src/main/java/io/apicurio/registry/connector/Compatibility.java:2]: 2020

**Comment by alesj** [comment]
lgtm
But like I wrote in the ConnectorApplication -- "Hack around TCCL switch -- keep in-sync with Kafka's ConnectDistributed::startConnect" :-)

**Comment by elakito** [comment]
There is a test error caused by NoClassDefError and I'll have to take a look if it is possible to avoid it...

**Comment by elakito** [comment]
The reason for java.lang.NoClassDefFoundError: org/apache/kafka/common/config/ConfigDef$CaseInsensitiveValidString seems to have been caused by the tests folder's pom that was pulling two different versions of kafka. The one directly pulled (version 2.3.0) and the other transitively pulled from apicurio-registry-utils-converter (now version 2.4.1, previously version 2.2.1). I added the exclusions so that kafka is only pulled directly. But there is still one test error.

**Comment by EricWittmann** [comment]
The one failing test is a flaky test and unrelated to your changes.  I've been meaning to exclude it in the **tests** module **pom.xml**.  I'm doing that now and will update this branch when it's working.

**Pull request merged**